src/module_wrap.h

```diff
@@ -38,6 +38,43 @@ enum ModulePhase : int {
   kEvaluationPhase = 2,
 };
 
+struct ModuleRequest : public MemoryRetainer {
+  using ImportAttributeVector =
+      std::vector<std::pair<std::string, std::string>>;
+
+  std::string specifier;
+  ImportAttributeVector import_attributes;
+  ModulePhase phase;
+
+  ModuleRequest(std::string specifier,
+                ImportAttributeVector import_attributes,
+                ModulePhase phase)
+      : specifier(specifier),
+        import_attributes(import_attributes),
+        phase(phase) {}
+
+  SET_MEMORY_INFO_NAME(ModuleRequest)
+  SET_SELF_SIZE(ModuleRequest)
+  void MemoryInfo(MemoryTracker* tracker) const override;
+
+  template <int elements_per_attribute = 3>
+  static ModuleRequest From(v8::Local<v8::Context> context,
+                            v8::Local<v8::String> specifier,
+                            v8::Local<v8::FixedArray> import_attributes,
+                            ModulePhase phase);
+  static ModuleRequest From(v8::Local<v8::Context> context,
+                            v8::Local<v8::ModuleRequest> v8_request);
+
+  bool operator<(const ModuleRequest& other) const {
+    if (specifier != other.specifier) {
+      return specifier < other.specifier;
+    } else if (import_attributes != other.import_attributes) {
+      return import_attributes < other.import_attributes;
+    }
```

Comment:
This comparison isn't clear to me at all - can you add a comment explaining where the comparator is and how it is determining comparison for import attributes?

In particular, how does this ensure that `{ type: 'a', other: 'attribute' }` is identical to `{ other: 'attribute', type: 'a' }`?

Reply:
It's a multi-key `std::less` variant. `std::map` use two `std::less` comparison on `lhs` and `rhs`, like `lhs < rhs === false && lhs > rhs === false` to determine equality.

Reply:
Reverted to `std::unordered_map`. So this is not needed.